pkg/receive/hashring.go

```diff
@@ -299,3 +294,14 @@ func HashringFromConfig(algorithm HashringAlgorithm, replicationFactor uint64, c
 
 	return newMultiHashring(algorithm, replicationFactor, config), err
 }
+
+func newHashring(algorithm HashringAlgorithm, endpoints []string, replicationFactor uint64) Hashring {
+	switch algorithm {
+	case AlgorithmHashmod:
+		return simpleHashring(endpoints)
+	case AlgorithmKetama:
+		return newKetamaHashring(endpoints, SectionsPerNode, replicationFactor)
+	default:
+		return simpleHashring(endpoints)
```

Comment:
If an invalid algorithm is specified, do we just fallback to the hashmod algorithm?
I feel a better way is to either: 
1. Return non-exist error and let users know this is wrong.
2. Fall back to the default one. But have a log to tell that the specified algorithm is wrong.

Those 2 options are better than ignoring the error.

Reply:
@yeya24 I agree with Option 1. Thought of that myself. But that would change the logic of [newMultiHashring()](https://github.com/thanos-io/thanos/blob/0d659bf171afa6bdf5c5ece3033df3a7e8245d8c/pkg/receive/hashring.go#L234-L248) a little bit. Would that cause a backward compatibility problem?

I see that we have an errors package at https://github.com/thanos-io/thanos/tree/main/pkg/errors but that doesn't seem to be used much? If we go with option 1, should I use that one or the `github.com/pkg/errors` one?

@fpetkovski @matej-g

Reply:
What would be the behavior though? Right now we do not return error as hashring creation (so to speak) "cannot" fail. Should we return error and ignore a hashring with invalid algorithm? Or how would we like to handle this?

This is why option 2 looked better, since we can just warn and fall back.

Reply:
When I look at this issue now, I feel either way is okay to me. 
If we go with option 1, even if it is wrong, it is just one-off. If we go with option 2, it is also fine to me as this is better than current behavior.